perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/ModulesYamlConfigLoader.java

```diff
@@ -0,0 +1,94 @@
+package cz.metacentrum.perun.core.impl.modules;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static cz.metacentrum.perun.core.impl.Utils.notNull;
+
+/**
+ * @author Vojtech Sassmann <vojtech.sassmann@gmail.com>
+ */
+
+public class ModulesYamlConfigLoader implements ModulesConfigLoader {
+
+	private final static Logger log = LoggerFactory.getLogger(ModulesYamlConfigLoader.class);
+
+	private static final ObjectMapper yamlObjectMapper = new ObjectMapper(new YAMLFactory());
+	private String modulesDirPath = "/etc/perun/modules/";
+
+	public ModulesYamlConfigLoader() { }
+
+	public ModulesYamlConfigLoader(String modulesDirPath) {
+		this.modulesDirPath = modulesDirPath;
+	}
+
+	@Override
+	public String loadStringProperty(String moduleName, String property) {
+		return loadPropertyNode(moduleName, property).asText();
+	}
+
+	@Override
+	public Integer loadIntegerProperty(String moduleName, String property) {
+		return loadPropertyNode(moduleName, property).isNull() ? null : loadPropertyNode(moduleName, property).asInt();
+	}
+
+	@Override
+	public List<String> loadStringListProperty(String moduleName, String property) {
+		List<String> values = new ArrayList<>();
+		JsonNode propertyNode = loadPropertyNode(moduleName, property);
+		propertyNode.iterator().forEachRemaining(node -> values.add(node.asText()));
+		return values;
+	}
+
+	@Override
+	public List<Integer> loadIntegerListProperty(String moduleName, String property) {
+		List<Integer> values = new ArrayList<>();
+		JsonNode propertyNode = loadPropertyNode(moduleName, property);
+		propertyNode.iterator().forEachRemaining(node -> values.add(node.isNull() ? null : node.asInt()));
+		return values;
+	}
+
+	private JsonNode loadPropertyNode(String moduleName, String propertyName) {
+		notNull(moduleName, "configFile");
+		notNull(propertyName, "propertyName");
+
+		String path = modulesDirPath + moduleName + ".yaml";
+
+		JsonNode root = loadModulesYamlFile(path);
+
+		JsonNode propertyNode = parsePropertyNode(root, propertyName);
+		if (propertyNode == null) {
```

Comment:
I am citing Pavel Z. who commented on this behavior in the previous PR:
"I just wonder, if we couldn't use empty value as a correct input for some logic. Like value is either set or not, then some logic performs specific or generic behavior. You could fall-back to default values by type like empty list / null / false etc."

Reply:
I have added new versions of the methods that allow you to specify a default value that is returned, if the found value is null or not found. I don't think it is a good idea to generally return some fallback default values. It might now be clear that it does return these default values and what they actually are.